lib/autocomplete-manager.coffee

```diff
@@ -110,26 +113,59 @@ class AutocompleteManager
     return if @isCurrentFileBlackListed()
     cursor = @editor.getLastCursor()
     return unless cursor?
-    cursorPosition = cursor.getBufferPosition()
-    currentScope = cursor.getScopeDescriptor()
-    return unless currentScope?
-    currentScopeChain = currentScope.getScopeChain()
-    return unless currentScopeChain?
-
-    options =
-      editor: @editor
-      buffer: @buffer
-      cursor: cursor
```

Comment:
Many providers in the wild have a need to determine a different prefix. This is almost uniformly done via a slightly different regex. If we were to provide the function to them, they wouldn't all have to reimplement that - this is an area with potentially large performance implications. The default function we have today is susceptible to it also. 

See https://github.com/atom-community/autocomplete-paths/commit/8bd05ed5724683eaed798e5b8c49d99b29ec68a2#diff-7d05e6fe4c61ffced4fb4bd806573aadR23 through https://github.com/atom-community/autocomplete-paths/commit/8bd05ed5724683eaed798e5b8c49d99b29ec68a2#diff-7d05e6fe4c61ffced4fb4bd806573aadR51.

The issue is that in large files with no line breaks, `cursor.getBeginningOfCurrentWordBufferPosition({wordRegex: @wordRegex})` will actually search all the way up to the last non-blank row (which is always line 0). This causes the regexp to perform like crap, and consequently, typing is noticeably affected.

Offending `cursor` code: https://github.com/atom/atom/blob/a3f64ddf7afdebfded53777638214e92b9f5937a/src/cursor.coffee#L505-L506